### PR TITLE
Created New PR For Config Hook

### PR DESCRIPTION
**Pull request by mexi-cano**
What (if any) features are you implementing?
- Created config hook
Any other comments, questions, or concerns?
- Created new PR. This PR branches from `dev` instead of `results_redesign`. It was simpler to create new PR w/ appropriate base than resolving the merge conflicts with the [latter open PR](https://github.com/Gary-Community-Ventures/benefits-calculator/pull/883)

### CODE REVIEW

**Comment by CalebPena** [comment]
@mexi-cano Can you screenshot what the raw api response looks like?

**Comment by mexi-cano** [comment]
> @mexi-cano Can you screenshot what the raw api response looks like?

Thanks for clarifying the concept. 

Screenshot 2024-01-31 at 1 48 32 PM


Screenshot 2024-01-31 at 1 51 16 PM

File: src/Components/Config/configHook.tsx [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  useEffect(() => {
+    getConfig().then((value: ConfigApiResponse[] | undefined) => {
+      // get data and set loading to false
+      try {
+        if (value !== undefined) {
+          const transformedOutput: Config = transformConfigResponse(value);
+          setConfigResponse(transformedOutput);
+        }
+      } catch (e) {
+        console.log(e);
```
Comment (CalebPena) [inline src/Components/Config/configHook.tsx:43]:
```suggestion
        console.error(e);
```

File: src/Components/Config/configHook.tsx [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    return response.json();
+  });
+}
+
+export default function useGetConfig() {
+  const [configLoading, setLoading] = useState<boolean>(true);
+  const [configResponse, setConfigResponse] = useState<Config | undefined>();
+
+  useEffect(() => {
+    getConfig().then((value: ConfigApiResponse[] | undefined) => {
```
Comment (CalebPena) [inline src/Components/Config/configHook.tsx:35]: I don't think that the api response can be undefined.
```suggestion
    getConfig().then((value: ConfigApiResponse[]) => {
```

**Review by CalebPena — APPROVED** [review]
Looks good.

File: src/Components/Config/configHook.tsx (hunk 2 above)
Comment (mexi-cano) [reply src/Components/Config/configHook.tsx]: Updated. thanks!

**Pull request merged**